src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/blob.py

```diff
@@ -0,0 +1,144 @@
+        The shared access signature used to access the source container. It is not required if
+        either a connection string is given or the source container doesn't require a sas.
+
+    :param str destination_container:
+        The destination blob container
+
+    :param str pattern:
+        If option --recursive is specified, then this command interprets the specified file pattern
+        as a blob prefix. If option --recursive is not specified, then this command matches the file
+        pattern against exact blob names.
+
+    :param bool recursive:
+        Copy all the files to the given container and maintain the folder structure.
+
+    :param bool snapshots:
+        Copy both the blobs and their snapshots.
+
+    :param bool exclude_old:
+        Excludes an older source resource. The resource will not be copied if the last modified time
+        of the source is the same or older than destination.
+
+    :param bool exclude_new:
+        Excludes a newer source resource. The resource will not be copied if the last modified time
+        of the source is the same or newer than destination.
+
+    :param str blob_type:
+        Specifies the type of the desination blob. the value can be block, page or append.
+
+    :return: A BlobCopyTicket instance summarize the operations
+    """
+
+    # TODO:
+    # 1. Page the result list (using num_result and marker)
+    # 2. Support connection string for source
+    # 3. stop using 'baseblobservice.exists' function. it doesn't provide performance gain
+    #    since it invoked the get_blob_properties any way.
+
+    # Question:
+    # 1. Performance of creating a source blob service
+
+    if pattern is None and not recursive:
+        # TODO: use a validator?
+        raise ValueError('Neither --recursive or --pattern is set.')
+
+    if exclude_old and exclude_new:
+        raise ValueError('Both --exclude-new and --exclude-old are set.')
+
+    src_client = BlockBlobService(account_name=source_account, sas_token=source_sas)
+
+    def _get_blob_name(source_blob):
+        name = source_blob.name
+        if source_blob.snapshot is not None:
+            # the snapshot time string has seven digital in the microseconds, which make strptime
+            # nearly unusable. therefore characters after dot are thrown away
+            time_string = source_blob.snapshot[:source_blob.snapshot.rfind('.')]
+            snapshot_time = datetime.strptime(time_string, '%Y-%m-%dT%H:%M:%S')
+
+            # insert the date time string before the file extension
+            dot = name.rfind('.')
+            dot = len(name) if dot == -1 else dot
+            name = '{0}({1}){2}'.format(name[0:dot],
+                                        snapshot_time.strftime('%Y-%m-%d %H%M%S'),
+                                        name[dot:])
+        return name
+
+    def _get_blob_url(source_blob):
+        src_url = src_client.make_blob_url(source_container, source_blob.name, sas_token=source_sas)
```

Comment:
This only needed because of the SDK limitation yes? If so, I recommend adding a TODO here to mention that this can be removed once the SDK is fixed and link to that GitHub issue